### PR TITLE
refactor(typescript): packages/rspack/src/util/StatsFactory

### PR DESCRIPTION
**Pull request by wxiaoyun**
## Summary

- Improve ts coverage
- Addresses https://github.com/web-infra-dev/rspack/issues/7225

## Checklist

- [X] Tests updated (or not required).
- [X] Documentation updated (or not required).

### CODE REVIEW

**Review by SoonIter — DISMISSED** [review]
LGTM, thanks❤️

File: packages/rspack/src/util/comparators.ts [hunk 1]
```diff
@@ -46,11 +46,12 @@ const concatComparatorsCache: TwoKeyWeakMap<
 	Comparator
 > = new TwoKeyWeakMap();
 
-export const concatComparators = (
+export function concatComparators(...comparators: Comparator[]): Comparator;
```
Comment (SoonIter) [inline packages/rspack/src/util/comparators.ts:49]: two solutions

1. type an array with length check
2. make `concatComparators` can accept only one argument

File: packages/rspack/src/util/comparators.ts [hunk 2]
```diff
@@ -46,11 +46,13 @@ const concatComparatorsCache: TwoKeyWeakMap<
 	Comparator
 > = new TwoKeyWeakMap();
 
+export type AtLeastTwo<T> = [T, T, ...T[]];
+
 export const concatComparators = (
-	c1: Comparator,
-	c2: Comparator,
-	...cRest: Comparator[]
+	...comps: AtLeastTwo<Comparator>
```
Comment (wxiaoyun) [inline packages/rspack/src/util/comparators.ts:52]: image

**Review by SoonIter — APPROVED** [review]
LGTM

**Pull request merged**